### ISSUE TITLE
Arango holds persistent connection with pod even after Node Death (Kubernetes)

### ISSUE THREAD

**Issue by joeltgray** [arangodb/arangodb#17734]
## My Environment

* __ArangoDB Version__:        3.8.7
* __Deployment Mode__:       Multinode
* __Deployment Strategy__:   Kubernetes
* __Configuration__:               3 Separate Nodes (VMs)
* __Infrastructure__:               Local Storage
* __Operating System__:        Ubuntu 20.04.3 LTS (GNU/Linux 5.4.0-126-generic x86_64)
* __Total RAM in your machine__:        48GB per Node
* __Disks in use__:        2x 300GB HDD per Node
* __Used Package__:    Docker - official Docker library

## Component, Query & Data

__Affected feature__:
ArangoDB hold a persistent connection with a pod/agent even after the host Node of that pod/agent is powered off.

__Dataset__:
Applies to any dataset, it is not data related. When creating a connection to ArangoDB and completing collection saves of any size/type Arango will hold a connection to the agent/coordinator/DB even if the host Node is off.

__Replication Factor & Number of Shards (Cluster only)__:
Replication Factor = 3

## Steps to reproduce

1. Create a Multinode Kubernetes System
2. Set up ArangoDB v3.8.7 with Replication Factor of 3.
3. Create a Python application using PyArango to connect to ArangoDB.
4. Create a loop saving changes to collections
5. Get the IP address of the Arango Coordinator Pod on Node 3: ```kubectl get pods -A -o wide -w| egrep 'arango' ```
6. Ensure Arango coordinator on Node 3 is being used for saving collections - using TCPDUMP:  ```sudo tcpdump -l -i any ip host {COORDINATOR_POD_IP} and dst port 8529 | awk '$5~/8529/{print $5}'```
7. Power off Node 3 and time how long it takes for the connection to fail.

__Problem__:
I have a Kubernetes Multinode system set up 3 three nodes. I am creating a connection between a pod on Node 2 to the Arango deployment using PyArango, the Arango Deployment has 3 coordinator pods one on Node 1, one on Node 2 and one on Node 3.

I'm testing out how resilient the system is and I've noticed an issue. It seems that if I'm updating collections on Arango and my program (running on Node 2) connects to the Arango Coordinator pod on Node 3 and I power off Node 3, the connection will not time out, it will simply stay put for as long as Kubernetes takes to sort it out itself by killing the pod.

But I want the connection to timeout if the connection is idle or getting no response after ~30 seconds.

I've tried some different things using the PyArango methods and the HTTP libraries and no luck. I dont think the issue lies with the PyArango library.

I've tried many different connection timeouts in my arango_cluster.yaml file for configuring ArangoDB at creation. See Below:

```
apiVersion: "database.arangodb.com/v1"
kind: "ArangoDeployment"
metadata:
  name: "mytest-arango"
spec:
  mode: Cluster
  image: arangodb/arangodb:3.8.7
  environment: Production
  tls:
    caSecretName: None
  auth:
    jwtSecretName: None
  externalAccess:
    type: None
  cluster:
    resources:
      requests:
        memory: 1800Mi
      limits:
        memory: 2Gi
  coordinators:
    args:
      - --http.keep-alive-timeout=5
      - --transaction.streaming-idle-timeout=5
      - --cluster.index-create-timeout=5
      - --server.authentication-timeout=5
      - --network.idle-connection-ttl=5
      - --query.max-runtime=5
  agents:
    args:
      - --http.keep-alive-timeout=5
      - --transaction.streaming-idle-timeout=5
      - --server.authentication-timeout=5
      - --network.idle-connection-ttl=5
      - --query.max-runtime=5
  dbservers:
    args:
      - --http.keep-alive-timeout=5
      - --transaction.streaming-idle-timeout=5
      - --server.authentication-timeout=5
      - --network.idle-connection-ttl=5
      - --query.max-runtime=5
```

__Expected result__:
I want ArangoDB to drop any IDLE or unresponsive connections after a few seconds, 30 seconds max.

**Comment by joeltgray** [comment]
Also just verified issue still exists in version 3.10.1. Any help much appreciated.

**Comment by joeltgray** [comment]
I've solved the problem.

In the same file where I make the conn = Connection(......) call. I needed to add a monkey patch to patch the AikidoSession class that PyArango uses to make the raw HTTP connections.

I added the following to the beginning of my file and now my connection times out correctly and doesn't hang for ages.

```
from pyArango.connection import AikidoSession

AikidoSession.Holder._base_call = AikidoSession.Holder.__call__
def _monkey_patch_call(self, *args, **kwargs):
  kwargs["timeout"] = 10
  return self._base_call(*args, **kwargs)

AikidoSession.Holder.__call__ = _monkey_patch_call
```

Although I don't understand why we need to go through this backwards loop to add a simple connection header.

Can we request directly specifying headers in the initial Connection class object instantiation?
